pilot/cmd/pilot-discovery/main.go

```diff
@@ -107,6 +107,8 @@ func init() {
 		"comma separated list of networking plugins to enable")
 
 	// Config Controller options
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.Config.InstallCRDs, "install-crds", false,
+		"Upon startup, Pilot will attempt to inspect the state of CRDs as installed in Kubernetes.  If it CRDs are not found, it will attempt to create them.  During CRD creation, if Pilot fails to verify creation, Pilot will attempt to clean-up after itself and delete CRDs.")
```

Comment:
Can you remove the 3rd sentence now that you removed the deregister logic in the error path below?

Reply:
Yes, thanks!